src/compiler/checker.ts

```diff
@@ -13379,9 +13379,9 @@ namespace ts {
                     }
                 }
                 else {
-                    if (languageVersion >= ScriptTarget.ES6 && !isInAmbientContext(node)) {
+                    if (compilerOptions.module === ModuleKind.ES6 && !isInAmbientContext(node)) {

```

Comment:
(languageVersion >= ScriptTarget.ES6 && !compilerOptions.module), or set that at the top of the file such as language version